src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/rewriting/rules/IdentifyLogicalOp.java

```diff
@@ -0,0 +1,50 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.rewriting.rules;
+
+import se.liu.ida.hefquin.engine.federation.FederationMember;
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.PhysicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+import se.liu.ida.hefquin.engine.queryplan.physical.PhysicalOperatorForLogicalOperator;
+
+public class IdentifyLogicalOp {
+
+    public static boolean matchJoin( final PhysicalOperator op ) {
+        final LogicalOperator lop = ((PhysicalOperatorForLogicalOperator) op).getLogicalOperator();
+        return lop instanceof LogicalOpJoin;
+    }
+
+    public static boolean matchTPAdd( final PhysicalOperator op ) {
+        final PhysicalOperatorForLogicalOperator lop = (PhysicalOperatorForLogicalOperator) op;
+        return lop.getLogicalOperator() instanceof LogicalOpTPAdd;
+    }
+
+    public static boolean matchBGPAdd( final PhysicalOperator op ) {
+        final PhysicalOperatorForLogicalOperator lop = (PhysicalOperatorForLogicalOperator) op;
+        return lop.getLogicalOperator() instanceof LogicalOpBGPAdd;
+    }
+
+    public static boolean matchUnion( final PhysicalOperator op ) {
+        final LogicalOperator lop = ((PhysicalOperatorForLogicalOperator) op).getLogicalOperator();
+        return lop instanceof LogicalOpUnion;
+    }
+
+    public static boolean matchMultiwayJoin( final PhysicalOperator op ) {
+        final LogicalOperator lop = ((PhysicalOperatorForLogicalOperator) op).getLogicalOperator();
+        return lop instanceof LogicalOpMultiwayJoin;
+    }
+
+    public static boolean matchMultiwayUnion( final PhysicalOperator op ) {
```

Comment:
I don't think the names of these functions are particularly suitable. I mean, if I would see them somewhere in the source code, I wouldn't immediately know what they are doing. Therefore, my proposal is to replace in these names the word "match" by the word "is"; e.g., `isTPAdd`.